pkg/devfile/parser/data/2.1.0/types.go

```diff
@@ -0,0 +1,435 @@
+package version210
+
+import "github.com/openshift/odo/pkg/devfile/parser/data/common"
+
+// CommandGroupType describes the kind of command group.
+// +kubebuilder:validation:Enum=build;run;test;debug
+type CommandGroupType string
+
+const (
+	BuildCommandGroupType CommandGroupType = "build"
+	RunCommandGroupType   CommandGroupType = "run"
+	TestCommandGroupType  CommandGroupType = "test"
+	DebugCommandGroupType CommandGroupType = "debug"
+)
+
```

Comment:
The changes that have been made to 2.0.0 by Enriquel should be moved to the 2.1.0 counterparts.

Reply:
If we are removing all mentions of alpha.build-dockerfile from the source code, then odo deploy will not have any way of detecting a dockerfile path from a devfile of version 2.0.0 anymore since it will expect a dockerfile component which won't be present in it
So are we planning to allow odo deploy to continue supporting 2.0.0 or switch to 2.1.0 completely?

Reply:
@pedjak  Your dockerfile component replaces `alpha.build-dockerfile` as the way to identify which dockerfile to use for build.    If dockerfile component isn't in the devfile, that would mean do not use docker build and it shouldn't use `alpha.build-dockerfile` as a backup.  The code as written right now just ignores `alpha.build-dockerfile` anyway.    And, 2.0.0 spec is done and should not be changed., right?  These dockerfile stuff are considered outer loop guidance that are new in 2.1.0.

Reply:
>The changes that have been made to 2.0.0 by Enriquel should be moved to the 2.1.0 counterparts. "ubild/deploy" outer loop enhancements are to be added to 2.1.0 rather than 2.0.0.

Not sure how to act on this and how it relates to the ```TestCommandGroupType``` and ```DebugCommandGroupType``` lines highlighted here exactly

Reply:
> So are we planning to allow odo d eploy to continue supporting 2.0.0 or switch to 2.1.0 completely?

Should we really have a 2.1.0 parser, I personally don't think so :) 

2.1.0 or 2.x would not have any breaking changes between 2.x and 2.(x-1) . Hence, we could "add" on to the existing devfile.

Rather, we should have a devfile 2 parser.